### DISCUSSION TITLE
Multiple Mask

### DISCUSSION THREAD

**Discussion by renerlemes** [uNmAnNeR/imaskjs#844] [Show and tell]
How to use two masks above:

1 - 000.000.000-00
2 - 00.000.000/0000-00

I try to use dispatch but no success.

Can anyone give an example please?

**Comment by schellerg** [comment]
> How to use two masks above:
> 
> 1 - 000.000.000-00 2 - 00.000.000/0000-00
> 
> I try to use dispatch but no success.
> 
> Can anyone give an example please?

```
mask: [
  {
    mask: '000.000.000-00',
  },
  {
    mask: '00.000.000/0000-00',
  },
],
```

**Reply by renerlemes** [reply]
This not work @schellerg

If I input 123.456.789-01 and input 2, the result should be 12.345.678/9012-

Always applies the first mask of array. When the input value is the same length of first value of array, the mask stop work!

**Reply by uNmAnNeR** [reply]
i cannot confirm the issue.
https://stackblitz.com/edit/javascript-3791yw?file=index.js

![Kapture 2024-06-02 at 11 24 46](https://github.com/uNmAnNeR/imaskjs/assets/1927921/9277ae5b-511a-4836-a894-0a1ce5f4ff9f)

**Reply by renerlemes** [reply]
Using your example works with UI. But if you use IMask.createMask, without UI, it will not work.

I try too using the following:

```
const masked = IMask.createMask({
  mask: [
    {
      mask: '000.000.000-00',
    },
    {
      mask: '00.000.000/0000-00',
    },
  ]
});

masked.resolve('123.456.789-012');

console.log(masked.value); //123.456.789-01
console.log(masked.unmaskedValue); //12345678901
```

https://codesandbox.io/p/sandbox/componentes-mask-nnzvfl?file=%2Fsrc%2Fcomponents%2FMask.tsx%3A36%2C34

**Reply by uNmAnNeR** [reply]
thanks for explanation. As a temporary solution you could use `masked.resolve('123.456.789-012', {});` but i will check if i can fix it in general

**Reply by renerlemes** [reply]
@uNmAnNeR the temporary solution worked! tks

Maybe it's the same problem, but when I try to use it with RegExp, it doesn't even let me type the mask.

The link with the problemn is the same
[https://codesandbox.io/p/sandbox/componentes-mask-nnzvfl?file=%2Fsrc%2FApp.tsx%3A20%2C31](https://codesandbox.io/p/sandbox/componentes-mask-nnzvfl?file=%2Fsrc%2FApp.tsx%3A20%2C31)

I try to put `mask: /^\d+-\d$/`, and the result would be 1-1, 12-3, 123-4, 1234-5...
